src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1354,6 +1396,29 @@ private void WriteRedSetters(Node node)
             }
         }
 
+        private TreeType GetBaseTypeWithField(TreeType node, string name)
+        {
+            TreeType bestType = null;
+            for (var current = node; current != null; current = TryGetBaseType(current))
+            {
+                var fields = GetNodeOrNodeListFields(current);
+                var field = fields.FirstOrDefault(f => f.Name == name);
```

Comment:
Nit: Is there a method to check if the list contains a field of the given name? We don't need to get the field back.
Nit: Consider renaming method `GetHighestBaseWithField` since we can return a type that isn't the node's `base`.

Reply:
Sure!